### DISCUSSION TITLE
Existence of unique solutions in AMPL

### DISCUSSION THREAD

**Discussion by TasPinar204** [coin-or/Bonmin#17] [General]
Dear Ladies and Gentlemen,

we are using AMPL, the AMPL API for Java and the Solver Bonmin to solve a nonlinear (quadratic) integer based problem. As stated in the theory of the optimization problem there may exist more than one solution. A question is whether it is possible that the solver gives out different solutions, when started with the same initial values several times?

Thank you for the support!
If you have any questions, don't hesitate to ask me.

Best Regards,

Pinar Tas

**Comment by svigerske** [comment]
As far as I remember, BONMIN is deterministic, so same input on same hardware should give same results, unless you use timelimits somewhere.
I don't think there is a feature build-in in BONMIN to look for "all" solutions.

**Comment by TasPinar204** [comment]
Hallo there,

thank you very much. This helped me a lot!

Best Regards!

Pinar Tas

Von: Stefan Vigerske [mailto:notifications@github.com]
Gesendet: Mittwoch, 13. Januar 2021 03:41
An: coin-or/Bonmin
Cc: Tas, Pinar (LfStat); Author
Betreff: Re: [coin-or/Bonmin] Existence of unique solutions in AMPL (#17)


As far as I remember, BONMIN is deterministic, so same input on same hardware should give same results, unless you use timelimits somewhere.
I don't think there is a feature build-in in BONMIN to look for "all" solutions.

**Reply by mjsaltzman** [reply]
If you are looking for alternate solutions, one procedure is to find the first optimum, then add a constraint that rules that solution out and re-solve. You can repeat the process until the optimal objective value changes, at which point you know you have exhausted the set of optima for the original problem.

**Reply by kkofler** [reply]
But that algorithm is not reliable for nonconvex problems, which can have local optima that are not globally optimal.

**Comment by TasPinar204** [comment]
Ok thank you very much.  With this information I could test it.

Best Regards!

Pinar Tas

Von: Matthew Saltzman [mailto:notifications@github.com]
Gesendet: Dienstag, 19. Januar 2021 16:32
An: coin-or/Bonmin
Cc: Tas, Pinar (LfStat); Author
Betreff: Re: [coin-or/Bonmin] Existence of unique solutions in AMPL (#17)


If you are looking for alternate solutions, one procedure is to find the first optimum, then add a constraint that rules that solution out and re-solve. You can repeat the process until the optimal objective value changes, at which point you know you have exhausted the set of optima for the original problem.

**Comment by TasPinar204** [comment]
Hi there,

we have a convex problem. So the global optimum is equal to the local optimum.

Best regards!

Pinar Tas

Von: Kevin Kofler [mailto:notifications@github.com]
Gesendet: Montag, 15. Februar 2021 02:21
An: coin-or/Bonmin
Cc: Tas, Pinar (LfStat); Author
Betreff: Re: [coin-or/Bonmin] Existence of unique solutions in AMPL (#17)


But that algorithm is not reliable for nonconvex problems, which can have local optima that are not globally optimal.
